packages/api/tests/lib/writeRecords/test-write-granules.js

```diff
@@ -683,6 +715,108 @@ test.serial('writeGranulesFromMessage() sets a default value of false for `publi
   t.is(translatedPgRecord.published, false);
 });
 
+test.serial('writeGranulesFromMessage() uses a default value for granule.createdAt from workflowStartTime if granule.createdAt is undefined', async (t) => {
+  const {
+    collectionCumulusId,
+    cumulusMessage,
+    executionCumulusId,
+    granuleId,
+    granuleModel,
+    knex,
+    providerCumulusId,
+    workflowStartTime,
+  } = t.context;
+
+  // Only test fields that are stored in Postgres on the Granule record.
+  // The following fields are populated by separate queries during translation
+  // or elasticsearch.
+  const omitList = ['files', '_id'];
+
+  // Remove createdAt key for test
+  delete cumulusMessage.payload.granules[0].createdAt;
+
+  await writeGranulesFromMessage({
+    cumulusMessage,
+    executionCumulusId,
+    providerCumulusId,
+    knex,
+    granuleModel,
+  });
+
+  const dynamoRecord = await granuleModel.get({ granuleId });
+  const granulePgRecord = await t.context.granulePgModel.get(
+    knex,
+    {
+      granule_id: granuleId,
+      collection_cumulus_id: collectionCumulusId,
+    }
+  );
+
+  // Validate objects all match
+  /// translate the PG granule to API granule to directly compare to Dynamo
+  const translatedPgRecord = await translatePostgresGranuleToApiGranule({
+    granulePgRecord,
+    knexOrTransaction: knex,
+  });
+  t.deepEqual(omit(translatedPgRecord, omitList), omit(dynamoRecord, omitList));
+
+  const esRecord = await t.context.esGranulesClient.get(granuleId);
+  t.deepEqual(omit(translatedPgRecord, omitList), omit(esRecord, omitList));
+
+  // Validate assertion is true in the primary datastore:
+
+  t.is(translatedPgRecord.createdAt, workflowStartTime);
+});
+
+test.serial('writeGranulesFromMessage() uses granule.createdAt value for written granule if defined', async (t) => {
```

Comment:
This is testing overwriting of `createdAt`, right? 🤔 If so, maybe can we update the test statement to make that explicit like the tests for `writeGranuleFromApi` do? 
```suggestion
test.serial('writeGranulesFromMessage() allows overwrite of createdAt and uses granule.createdAt value for written granule if defined', async (t) => {
```